services/proctoring-service/ai-service/main.py

```
@@ -57,42 +88,92 @@ def extract_features(frame, last_gray, head_angles, yolo_results):
+        maybe_save_frame(frame)
+
+        detected_events = []
+        yolo_results = yolo_model.predict(frame, conf=0.4, verbose=False)[0]
+        head_angles = None
+        fm_results = mp_face_mesh.process(cv2.cvtColor(frame, cv2.COLOR_BGR2RGB))
+        if fm_results.multi_face_landmarks and len(fm_results.multi_face_landmarks) == 1:
+            head_angles = get_head_pose(fm_results.multi_face_landmarks[0].landmark, frame.shape)
+
+        # Trích xuất và áp dụng luật
+        feats = extract_features(frame, None, head_angles, yolo_results)
+        if feats["person_count"] == 0: detected_events.append({"event_type": "FACE_NOT_DETECTED", "severity": "high", "metadata": {}})
```

Ruff (0.14.4)

Multiple statements on one line (colon)

(E701)